### PR TITLE
TOOLS-2824: Add static analysis task that runs `evergreen validate`

### PR DESCRIPTION
**Pull request by edobranov**
Jira: https://jira.mongodb.org/browse/TOOLS-2824

Evergreen patch with task: https://evergreen.mongodb.com/version/606f6d6cd6d80a5fbdd77f56

This adds a task for `evergreen validate` in the buildscript and updates our Static Analysis buildvariant with it. Also adding some sample output of different scenarios below.

Everything succeeds:
```
$ go run build.go -v sa:evgvalidate
START  | sa:evgvalidate
       | exec: '.../bin/evergreen validate --file common.yml'
FINISH | sa:evgvalidate in 971.621181ms
---------------
Completed in 971.889045ms
```

Breaking the syntax:
```
$ go run build.go -v sa:evgvalidate
START  | sa:evgvalidate
       | exec: '.../bin/evergreen validate --file common.yml'
FAIL   | sa:evgvalidate in 807.488006ms
       | error from `evergreen validate`: ERROR: load project error(s): error unmarshalling into parser project: yaml: line 282: did not find expected key
       | common.yml is an invalid configuration: exit status 1
task(s) [sa:evgvalidate] failed
exit status 2
```

Misspelling a distro name:
```
$ go run build.go -v sa:evgvalidate
START  | sa:evgvalidate
       | exec: '.../bin/evergreen validate --file common.yml'
FAIL   | sa:evgvalidate in 863.701372ms
       | error from `evergreen validate`: WARNING: yaml: unmarshal errors:
       |   line 302: key "mongod_port" already set in map
       |   line 294: key "mongod_port" already set in map
       |   line 305: key "mongod_port" already set in map
       | WARNING: buildvariant 'suse12' in project '' references a nonexistent distro 'susie12-sp5-small'.
       | common.yml is valid with warnings
task(s) [sa:evgvalidate] failed
exit status 2
```
^ for the last one, I'll make a comment below about `key "mongod_port" already set in map`

### CODE REVIEW

File: buildscript/sa.go
Comment (edobranov) [inline buildscript/sa.go:60]: Since there's some shortcomings with the parser detailed in [EVG-14227](https://jira.mongodb.org/browse/EVG-14227), I'm thinking of making a TOOLS ticket blocked on that EVG ticket for removing the duplicate key special-casing below (I'll add it once everything is lgtm'd).

The special-casing is just looping through the `evergreen validate` output and only returns an error if it sees a warning that's not associated with duplicate keys. It shouldn't break when the Evergreen team updates to `yaml.v3`, since these types of warnings will just disappear.

File: common.yml
Comment (edobranov) [inline common.yml:10]: This should've been `variables` since `mongo_tools_variables` isn't a valid keyword to define on the root.

See [this line](https://github.com/evergreen-ci/evergreen/blob/a4d568801804ccc151e65fcaaaa00e3886be9fbd/self-tests.yml#L40) of Evergreen's config file.

File: common.yml
Comment (edobranov) [inline common.yml:346]: We need this for the `SAEvergreenValidate` func to run the `evergreen` binary, which is found in the `$HOME` directory on distros.

Comment (rychipman) [reply common.yml:346]: perhaps we can do this in `_set_shell_env` instead?

File: buildscript/sa.go
Comment (rychipman) [reply buildscript/sa.go:60]: > I'm thinking of making a TOOLS ticket blocked on that EVG ticket for removing the duplicate key special-casing below

I think this is a good idea. I'd say just go ahead and file it now; no need to wait for this to get LGTM'ed

**Review by rychipman — CHANGES_REQUESTED** [review]

File: common.yml
Comment (edobranov) [reply common.yml:346]: Hmm, I tried simply adding `export PATH=$PATH:$HOME` at the bottom of `_set_shell_env` in [this patch build](https://evergreen.mongodb.com/task/mongo_tools_static_evergreen_validate_patch_cbdf89967b93b03a915e87d1c7694f77671c4769_6070a3f89ccd4e38a8c71257_21_04_09_18_59_11) (with the [common.yml diff](https://evergreen.mongodb.com/filediff/6070a3f89ccd4e38a8c71257/?file_name=common.yml&patch_number=0)), but it's somehow giving me a Go import cycle error.

I also tried it without `export` but it's the [same result](https://evergreen.mongodb.com/task/mongo_tools_static_evergreen_validate_patch_cbdf89967b93b03a915e87d1c7694f77671c4769_6070ad85c9ec440a16248859_21_04_09_19_39_54) (with the [common.yml diff](https://evergreen.mongodb.com/filediff/6070ad85c9ec440a16248859/?file_name=common.yml&patch_number=0)). Not totally sure why that's happening...

I was originally thinking that `_set_shell_env` was more for Evergreen expansions, but I'm fine exporting the PATH there if you have a sense of what could be causing that error.

**Review by rychipman — APPROVED** [review]
LGTM

**Pull request merged**
